src/vs/workbench/contrib/preferences/browser/settingsList.ts

```diff
@@ -0,0 +1,437 @@
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { ISettingsEditorViewState, SettingsTreeElement, SettingsTreeGroupElement, SettingsTreeSettingElement } from 'vs/workbench/contrib/preferences/browser/settingsTreeModels';
+import { ITreeRenderer } from 'vs/base/browser/ui/tree/tree';
+import { isDefined, isUndefinedOrNull } from 'vs/base/common/types';
+import { SettingsTreeDelegate, ISettingItemTemplate, SettingsTreeFilter } from 'vs/workbench/contrib/preferences/browser/settingsTree';
+import { focusBorder, foreground, errorForeground, inputValidationErrorBackground, inputValidationErrorForeground, inputValidationErrorBorder, scrollbarSliderHoverBackground, scrollbarSliderActiveBackground, scrollbarSliderBackground, editorBackground, editorForeground } from 'vs/platform/theme/common/colorRegistry';
+import { RGBA, Color } from 'vs/base/common/color';
+import { settingsHeaderForeground } from 'vs/workbench/contrib/preferences/browser/settingsWidgets';
+import 'vs/css!./media/settingsListScrollbar';
+import { localize } from 'vs/nls';
+import { Button } from 'vs/base/browser/ui/button/button';
+import { attachButtonStyler } from 'vs/platform/theme/common/styler';
+
+const $ = DOM.$;
+
+class SettingsListPaginator {
+	readonly PAGE_SIZE = 20;
+
+	private settings: SettingsTreeSettingElement[] = [];
+	private page = 1;
+
+	get currentPage(): number {
+		return this.page;
+	}
+
+	get totalPages(): number {
+		return Math.ceil(this.settings.length / this.PAGE_SIZE);
+	}
+
+	get settingsOnPage(): SettingsTreeSettingElement[] {
+		return this.settings.slice(
+			(this.page - 1) * this.PAGE_SIZE,
+			this.page * this.PAGE_SIZE,
+		);
+	}
+
+	constructor(private onPageChange: (shouldScroll: boolean) => void) { }
+
+	setSettings(settings: SettingsTreeSettingElement[], scrollToPage?: number): void {
+		this.settings = settings;
+		this.setPage(scrollToPage ?? this.page);
+	}
+
+	nextPage(): void {
+		const nextStartIdx = this.page * this.PAGE_SIZE;
+
+		if (this.settings.length > nextStartIdx) {
+			this.setPage(this.page + 1, true);
+		}
+	}
+
+	previousPage(): void {
+		if (this.page > 1) {
+			this.setPage(this.page - 1, true);
+		}
+	}
+
+	setPage(page: number, shouldScroll = true): void {
+		if (1 <= page && page <= this.totalPages) {
+			this.page = page;
+			this.onPageChange(shouldScroll);
+		}
+	}
+}
+
+interface ISettingsListView {
+	group: SettingsTreeGroupElement;
+	settings: SettingsTreeSettingElement[];
+	focusedSetting?: SettingsTreeSettingElement;
+}
+
+interface ISettingsListCacheItem {
+	container: HTMLElement;
+	template: ISettingItemTemplate;
+}
+
+export class SettingsList extends Disposable {
+	private searchFilter: (element: SettingsTreeElement) => boolean;
+	private getTemplateId = new SettingsTreeDelegate().getTemplateId;
```

Comment:
Not a safe way to access a method on an instance. You would have to call `.bind()` with the instance. But you should just hang on to the SettingsTreeDelegate itself